src/main/java/it/pagopa/pn/deliverypush/action/utils/TimelineUtils.java

```diff
@@ -1071,18 +1129,24 @@ public TimelineElementInternal buildCancelRequestTimelineElement(NotificationInt
             EventId.builder()
                 .iun(notification.getIun())
                 .build());
-        CancellationRequestDetailsInt details = CancellationRequestDetailsInt.builder().build();
+        CancellationRequestDetailsInt details = CancellationRequestDetailsInt.builder().
+            cancellationRequestId(UUID.randomUUID().toString()).
+            build();
         return buildTimeline(notification, TimelineElementCategoryInt.NOTIFICATION_CANCELLATION_REQUEST, elementId, details);
     }
 
     public TimelineElementInternal buildCancelledTimelineElement(NotificationInt notification){
         log.debug("buildCancelRequestTimelineElement - IUN={}", notification.getIun());
 
+        List<Integer> notRefined = notRefinedRecipientIndexes(notification);
         String elementId = TimelineEventId.NOTIFICATION_CANCELLED.buildEventId(
             EventId.builder()
                 .iun(notification.getIun())
                 .build());
-        CancelledDetailsInt details = CancelledDetailsInt.builder().build();
+        CancelledDetailsInt details = CancelledDetailsInt.builder().
+            notificationCost(100 * notRefined.size()).
```

Comment:
espliciterei meglio il "100" con una costante, che dovrebbe pure esser già presente in
it.pagopa.pn.deliverypush.service.impl.NotificationProcessCostServiceImpl#PAGOPA_NOTIFICATION_BASE_COST

Reply:
ehm, è ancora 100 * ....
l'hai modificato solo sopra :)